pkg/cmd/stack/deploy.go

```diff
@@ -187,14 +187,14 @@ func deploy(ctx context.Context, s *model.Stack, c kubernetes.Interface, config
 
 			// add specific stack labels
 			if _, ok := endpoint.Labels[model.StackNameLabel]; !ok {
-				endpoint.Labels[model.StackNameLabel] = s.Name
+				endpoint.Labels[model.StackNameLabel] = format.ResourceK8sMetaString(s.Name)
 			}
 			if _, ok := endpoint.Labels[model.StackEndpointNameLabel]; !ok {
 				endpoint.Labels[model.StackEndpointNameLabel] = endpointName
 			}
 
 			translateOptions := &ingresses.TranslateOptions{
-				Name:      s.Name,
+				Name:      format.ResourceK8sMetaString(s.Name),
```

Comment:
In line 391 of this file we are also creating the same type (`ingresses.TranslateOptions`) but we are not sanitizing the name, I guess we should do it in both places